### PR TITLE
Add dockerfiles

### PR DESCRIPTION
**Pull request by mehmetcanay**
Addition of Dockerfiles for frontend and backend.

### CODE REVIEW

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        npm install --legacy-peer-deps
+        npm ci --loglevel verbose --userconfig npmrc --max-old-space-size=16096 -legacy-peer-deps
+        npm run build --userconfig npmrc --max-old-space-size=16096
+    - name: Build & push frontend
+      uses: docker/build-push-action@v5
+      with:
+        context: ./frontend
+        file: ./frontend/Dockerfile
+        push: true
+        tags: ghcr.io/${{ steps.lowercase.outputs.REPO }}/frontend:latest-main, ghcr.io/${{ steps.lowercase.outputs.REPO }}/frontend:${{ steps.versions.outputs.FRONTEND_VERSION }}-main
```
Comment (tiadams) [inline .github/workflows/docker-image.yml:50]: omit the -main suffix

File: .github/workflows/docker-image.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    - name: Build & push backend
+      uses: docker/build-push-action@v5
+      with:
+        context: ./backend
+        file: ./backend/Dockerfile
+        push: true
+        tags: ghcr.io/${{ steps.lowercase.outputs.REPO }}/backend:latest-main, ghcr.io/${{ steps.lowercase.outputs.REPO }}/backend:${{ steps.versions.outputs.BACKEND_VERSION }}-main
```
Comment (tiadams) [inline .github/workflows/docker-image.yml:57]: same here

**Review by tiadams — CHANGES_REQUESTED** [review]

File: .github/workflows/docker-image.yml (hunk 1 above)
Comment (mehmetcanay) [reply .github/workflows/docker-image.yml]: omitted

**Review by tiadams — APPROVED** [review]
LGTM

**Pull request merged**